_delphi_utils_python/delphi_utils/validator/dynamic.py

```diff
@@ -241,8 +241,11 @@ def create_dfs(self, geo_sig_df, api_df_or_error, checking_date, geo_type, signa
         recent_lookbehind = timedelta(days=1)
 
         # semirecent_lookbehind: starting from the check date and working backward
-        # in time, how many days do we use to form the reference statistics.
-        semirecent_lookbehind = timedelta(days=7)
+        # in time, how many days do we use to form the reference statistics,
+        # with 7 as the minimum
+        semirecent_lookbehind = timedelta(days= max(
+            14, self.params.max_check_lookbehind
+            ))
```

Comment:
Were you meaning to use 14 instead of 7 here? Since the minimum is meant to be a threshold on the smallest number of days of reference data that is reasonable/useful to use, it needs to be enough data to calculate a standard deviation (>1 data point) + some additional days to be "useful". Do you think 7 is too low for this? Just wondering the reasoning behind changing the value.

Reply:
I thought that using 14 instead of 7 was one of the tests we did because we were worried that 7 days may be too short of a period. I think that particular set of results weren't super conclusive, so we can keep at 7 too - no real strong opinions here

Reply:
The 7-day default reference period and the 7-day minimum reference period are actually separate settings, that is, they don't *have* to have the same value. My understanding was that we increased the default reference period (the value `max_check_lookbehind` has if the params file doesn't set a value for `ref_window_size`) from 7 to 14 days.

The minimum reference period on the other hand is only used if the user sets `ref_window_size` to a really small value (3 or -10 or whatever). We don't really expect the minimum to get used in production. I was imagining it more as a reasonable override if someone was running this locally and was playing around with the params file.

> no real strong opinions here

Agree (just wanted to verify -- thanks for clarifying). Changing the min either way has no impact on how this behaves in production, so whatever you want to do here is fine.

Reply:
Ah, so what you're saying is make the default 14, but the minimum 7. That can be done too!